src/shared/CMakeLists.txt

```diff
@@ -323,25 +325,25 @@ GET_SHARED_FILES(clucene_shared_Files ".")
 
 #find our headers
 file(GLOB_RECURSE HEADERS ${clucene-shared_SOURCE_DIR}/*.h)
-
-add_library(clucene-shared SHARED
-	${clucene_shared_Files} ${HEADERS}
-)
-
-#set properties on the libraries
-SET_TARGET_PROPERTIES(clucene-shared PROPERTIES
-    VERSION ${CLUCENE_VERSION}
-    SOVERSION ${CLUCENE_SOVERSION}
-    COMPILE_DEFINITIONS_DEBUG _DEBUG
-)
-TARGET_LINK_LIBRARIES(clucene-shared ${CMAKE_THREAD_LIBS_INIT})
-IF ( EXTRA_LIBS )
-  TARGET_LINK_LIBRARIES(clucene-shared ${EXTRA_LIBS})
-ENDIF ( EXTRA_LIBS )
-
-install(TARGETS clucene-shared 
-        DESTINATION ${LIB_DESTINATION} 
-        COMPONENT development )
+IF ( BUILD_SHARED_LIBRARIES )
```

Comment:
we can tolerant this build shared libraries by default, and do the additional purge in the install path.

Reply:
> we can tolerant this build shared libraries by default, and do the additional purge in the install path.

But is it necessary to reserve shared libraries by default? in fact, we will only use static libraries, and adding this config condition will not break the existing build logic.

Reply:
it doesn't break anything, but if the goal can be achieved by not changing the code, why not change it anyway?